account_loan/models/account_loan_line.py

```diff
@@ -0,0 +1,447 @@
+                "credit": 0,
+                "debit": self.payment_amount - self.interests_amount,
+            }
+        )
+        if self.long_term_loan_account_id and self.long_term_principal_amount:
+            vals.append(
+                {
+                    "account_id": self.loan_id.short_term_loan_account_id.id,
+                    "credit": self.long_term_principal_amount,
+                    "debit": 0,
+                }
+            )
+            vals.append(
+                {
+                    "account_id": self.long_term_loan_account_id.id,
+                    "credit": 0,
+                    "debit": self.long_term_principal_amount,
+                }
+            )
+        return vals
+
+    def _invoice_vals(self):
+        return {
+            "loan_line_id": self.id,
+            "loan_id": self.loan_id.id,
+            "move_type": "in_invoice",
+            "partner_id": self.loan_id.partner_id.id,
+            "invoice_date": self.date,
+            "journal_id": self.loan_id.journal_id.id,
+            "company_id": self.loan_id.company_id.id,
+            "invoice_line_ids": [(0, 0, vals) for vals in self._invoice_line_vals()],
+        }
+
+    def _invoice_line_vals(self):
+        vals = list()
+        vals.append(
+            {
+                "product_id": self.loan_id.product_id.id,
+                "name": self.loan_id.product_id.name,
+                "quantity": 1,
+                "price_unit": self.principal_amount,
+                "account_id": self.loan_id.short_term_loan_account_id.id,
+            }
+        )
+        vals.append(
+            {
+                "product_id": self.loan_id.interests_product_id.id,
+                "name": self.loan_id.interests_product_id.name,
+                "quantity": 1,
+                "price_unit": self.interests_amount,
+                "account_id": self.loan_id.interest_expenses_account_id.id,
+            }
+        )
+        return vals
+
+    def _generate_move(self):
+        """
+        Computes and post the moves of loans
+        :return: list of account.move generated
+        """
+        res = []
+        for record in self:
+            if not record.move_ids:
+                if record.loan_id.line_ids.filtered(
+                    lambda r: r.date < record.date and not r.move_ids
+                ):
+                    raise UserError(_("Some moves must be created first"))
+                move = self.env["account.move"].create(record._move_vals())
+                move.action_post()
+                res.append(move.id)
+        return res
+
+    def _long_term_move_vals(self):
+        return {
+            "loan_line_id": self.id,
+            "loan_id": self.loan_id.id,
+            "date": self.date,
+            "ref": self.name,
+            "journal_id": self.loan_id.journal_id.id,
+            "line_ids": [(0, 0, vals) for vals in self._get_long_term_move_line_vals()],
```

Comment:
```suggestion
            "line_ids": [Command.create(vals) for vals in self._get_long_term_move_line_vals()],
```

Reply:
Not necessary, but cleaner this way :wink:  Thanks!